src/models/tools/geometry/geometry-content.ts

```diff
@@ -32,11 +33,100 @@ export function defaultGeometryContent(overrides?: JXGProperties) {
   return GeometryContentModel.create({ changes: [changeJson] });
 }
 
+// track selection in metadata object so it is not saved to firebase but
+// also is preserved across document/content reloads
+export const GeometryMetadataModel = types
+  .model("GeometryMetadata", {
+    id: types.string,
+    selection: types.map(types.boolean)
+  })
+  .views(self => ({
+    isSelected(id: string) {
+      return !!self.selection.get(id);
+    },
+    hasSelection() {
+      let hasSelection = false;
+      // TODO: short-circuit after first true
+      self.selection.forEach(value => {
+        if (value) {
+          hasSelection = true;
+        }
+      });
+      return hasSelection;
+    }
+  }))
+  .actions(self => ({
+    select(id: string) {
+      self.selection.set(id, true);
+    },
+    deselect(id: string) {
+      self.selection.set(id, false);
+    }
+  }));
+export type GeometryMetadataModelType = Instance<typeof GeometryMetadataModel>;
+
+function setElementColor(board: JXG.Board, id: string, selected: boolean) {
+  const element = board.objects[id];
+  if (element) {
+    element.setAttribute({
+              fillColor: selected ? kPointDefaults.selectedFillColor : kPointDefaults.fillColor,
+              strokeColor: selected ? kPointDefaults.selectedStrokeColor : kPointDefaults.strokeColor
+            });
+  }
+}
+
 export const GeometryContentModel = types
   .model("GeometryContent", {
     type: types.optional(types.literal(kGeometryToolID), kGeometryToolID),
     changes: types.array(types.string)
   })
+  .volatile(self => ({
+    metadata: undefined as any as GeometryMetadataModelType
```

Comment:
What's this do?

Reply:
`metadata` starts off `undefined` and is then set to `GeometryMetadataModelType` in `doPostCreate()`. We could declare it `GeometryMetadataModelType | undefined`, but then all usage would have to handle the undefined case (or use `!`). This just casts it so that it can always be treated as a `GeometryMetadataModelType`.